lib/mu_search/index_manager.rb

```diff
@@ -40,21 +39,13 @@ def fetch_indexes(type_name, allowed_groups, force_update: false)
       @master_mutex.synchronize do
         type_names.each do |type_name|
           if allowed_groups
-            if @configuration[:additive_indexes]
-              additive_indexes = find_indexes_for_groups type_name, allowed_groups
-              indexes_to_update += additive_indexes
-              @logger.debug("INDEX MGMT") do
-                index_names_s = additive_indexes.map(&:name).join(", ")
-                "Fetched #{additive_indexes.length} additive indexes for type '#{type_name}' and allowed_groups #{allowed_groups}: #{index_names_s}\nMatching allowed groups: #{additive_indexes.map(&:allowed_groups).to_json}"
-              end
-            else
-              index = get_combined_index_for_groups type_name, allowed_groups
-              unless index.nil?
-                indexes_to_update << index
-                @logger.debug("INDEX MGMT") { "Fetched index for type '#{type_name}' and allowed_groups #{allowed_groups}: #{index.name}" }
-              end
+            additive_indexes = find_indexes_for_groups type_name, allowed_groups
+            indexes_to_update += additive_indexes
+            @logger.debug("INDEX MGMT") do
+              index_names_s = additive_indexes.map(&:name).join(", ")
+              "Fetched #{additive_indexes.length} additive indexes for type '#{type_name}' and allowed_groups #{allowed_groups}: #{index_names_s}\nMatching allowed groups: #{additive_indexes.map(&:allowed_groups).to_json}"
             end
-          elsif @indexes[type_name] # fetch all indexes, regardless of access rights
+          elsif @indexes[type_name] # fetch all indexes, regardless of access rights. Only used in a stack without mu-authorization.
```

Comment:
I don't think this comment holds. Afaik it's used when using "admin" routes such as delete, where you can delete all indexes for a certain type

Reply:
You're right. We checked the search-endpoint, where we use `get_allowed_groups_with_fallback` to get the allowed groups (from the incoming request with a fallback to execute a dummy SPARQL query and getting the groups from that response).  In that case you will only end without allowed groups if mu-authorization isn't  part of the stack.
But for the 'admin' endpoints we use `get_allowed_groups` instead of  `get_allowed_groups_with_fallback`. Will remove the comment and document in the README that those admin endpoints should not be publicly exposed.